### PR TITLE
Refactor most of the places extracting polygons from geojson files

### PR DESCRIPTION
**Pull request by dabreegster**
The copy-pasted code was getting out of hand. I left `importer/src/bin/geojson_to_osmosis.rs` and `importer/src/bin/pick_geofabrik.rs` alone, because they don't want a map-space `Polygon`.

There shouldn't be any behavioral changes; I ran some import steps and verified lack of diff.

### CODE REVIEW

File: importer/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ collisions = { path = "../collisions" }
 convert_osm = { path = "../convert_osm" }
 csv = "1.1.4"
 geo = "0.17.0"
-geojson = "0.22"
+geojson = { version = "0.22.0", features = ["geo-types"] }
```
Comment (dabreegster) [inline importer/Cargo.toml:20]: This feature is being used in this crate, but previously it was being silently added from `popdat`

**Review by michaelkirk — APPROVED** [review]
LGTM

File: importer/Cargo.toml (hunk 1 above)
Comment (dabreegster) [reply importer/Cargo.toml:20]: https://blog.rust-lang.org/2021/05/11/edition-2021.html#default-cargo-feature-resolver
I wasn't, but awesome! Maybe we ought to opt into some of this sooner.

File: geom/src/polygon.rs [hunk 2]
```diff
@@ -603,3 +621,55 @@ fn downsize(input: Vec<usize>) -> Vec<u16> {
...
+        _ => anyhow::bail!("Unexpected geojson: {:?}", geojson),
+    };
+
+    let mut results = Vec::new();
+    for feature in features {
+        if let Some(geom) = &feature.geometry {
+            let raw_pts = match &geom.value {
+                geojson::Value::Polygon(pts) => pts,
+                // If there are multiple, just use the first
+                geojson::Value::MultiPolygon(polygons) => &polygons[0],
```
[the remark this answers is not in the document]
Comment (dabreegster) [reply geom/src/polygon.rs:644]: For the few uses we have so far (actdev study areas, UK census tracts, procedurally generated buildings), it works. What's the more reasonable thing to do with multipolygons -- duplicate the tags and return a result for each of the inner polygons? Probably... but I guess it'd be easier to decide with an example. I don't encounter multipolygons in the wild too often.

File: convert_osm/src/lib.rs [hunk 3]
```diff
@@ -145,35 +145,10 @@ fn use_amenities(map: &mut RawMap, amenities: Vec<(Pt2D, Amenity)>, timer: &mut
...
-                }
-            }
-        }
-    }
-
-    // Add these as new buildings, generating a new dummy OSM ID.
+    let force_convert = false;
     let mut id = -1;
-    for polygon in polygons {
+    for (polygon, _) in Polygon::from_geojson_file(path, &map.gps_bounds, force_convert)? {
```
[the remark this answers is not in the document]
Comment (dabreegster) [reply convert_osm/src/lib.rs]: Great point; I tried fixing this last night, but my tired brain also couldn't understand what `force_convert` was meant to be and how to invert it. I switched to `require_in_bounds`.

**Pull request merged**